pyproject.toml

```diff
@@ -31,9 +31,15 @@ tinkerforge = ["tinkerforge"]
 
 [tool.poetry.group.dev.dependencies]
 jupyter = "^1.0.0"
-pytest = "^7.2.0"
 ipython = "^8.4.0"
 ipywidgets = "^7.7.1"
+flake8 = "^5.0.4"
```

Comment:
it looks like version 6.0.0 is now available (https://pypi.org/project/flake8/#history); do we want to update?

Reply:
Yes! Let's do that whilst we're updating the pre-commit tools anyway. 

- There don't seem to be any code changes we need to include to pass the flake8 tests under version 6. 
- We do need to specify the minimum python version to be 3.8.10 (up from 3.8.0) but that's not a problem.
- That leads to changes in the lock file.